src/init.cpp

```diff
@@ -1043,6 +1044,8 @@ bool AppInitParameterInteraction()
         return InitError(strprintf("acceptnonstdtxn is not currently supported for %s chain", chainparams.NetworkIDString()));
     nBytesPerSigOp = GetArg("-bytespersigop", nBytesPerSigOp);
 
+    nDustLimit = GetArg("-dustlimit", nDustLimit);
```

Comment:
```suggestion
    if (IsArgSet("-dustlimit"))
    {
        CAmount n = 0;
        if (!ParseMoney(GetArg("-dustlimit", ""), n))
            return InitError(AmountErrMsg("dustlimit", GetArg("-dustlimit", "")));
        nDustLimit = n;
    }
```
This will introduce checks like non-negative.